modules/tools/pages/udfs-ui.adoc

```diff
@@ -0,0 +1,70 @@
+
+Start at the administration console and select menu:Query[UDFs] from the menus to access the `UDF` screen:
+
+image::udfs-ui/udfs-screen.png[alt="screen for adding user-defined functions"]
+
+== Javascript Function Libraries
+
+Click on `{plus}{nbsp}add function library` link to create a library of JavaScript functions. 
+
+image::udfs-ui/add-library-screen.png[,500, alt="add javascript library screen"]
+
+When you create a new library, an example function (`add`) is supplied as an example. 
+You can delete it when you add your own function, or leave it to serve as a basic example.
+
+You can set a `Namespace` for your library which will restrict its access to users that have permissions to access the bucket and scope defined in the Namespace.
+
+Before you leave the screen, you must set a `Library Name`.
+
+Press the btn:[Save] to leave the editor and return to the main UDF screen.
+
+image::udfs-ui/udf-screen-with-library-added.png[]
+
+You can `edit` the library again to add and/or delete functions contained inside it. 
+
+You can also delete the whole library by clicking on the `Drop` link.
+
+[#user-defined-functions]
+== User-Defined Functions
+
+You can add your own user-defined functions in the lower `User-Defined Functions` list. These functions can be one of two types:
+
+[horizontal]
+*Inline*:: This is a function written in {sqlpp} which can be used as part of another {sqlpp} statement (such as xref:n1ql:n1ql-language-reference/selectintro.adoc[`SELECT`] and xref:n1ql:n1ql-language-reference/execfunction.adoc[`EXECUTE FUNCTION`])
+
+*JavaScript*:: You can create {sqlpp} functions that call the Javascript functions defined in your library.
+
+To add a user-defined function, click on `{plus}{nbsp}add function` below the `User-Defined Functions` list.
+
+image::udfs-ui/add-function.png[,500, alt="Add function screen"]
+
+The `Namespace` can be set to define the scope that function resides in (`travel-sample.inventory` for example). 
+
+Users will need permissions to access this namespace to run the function.
+
+Use any valid identifier as the `Function Name`.
+The name must be unique and is case-insensitive.
+
+You can define a list of fixed parameters for your function, or you can use the `...` symbol which indicates a variable length function list defined as `args[]`.
+
+As noted <<user-defined-functions, above>>, the `Function Type` can be either `inline` or `Javascript`. 
+
+If you are defining an `inline` function then you can fill in any valid {sqlpp} expression which can then be used as part of another {sqlpp} statement.
+This is the default when creating a function.
+
+If you switch `Function Type` to `Javascript` then the dialog will change to configure a JavaScript function call:
+
+image::udfs-ui/add-javascript-function.png[,500, alt="add Javascript function"]
+
+Then you select the `Javascript Library` where you created your Javascript function, and the name of your function. (There is no need to include the parameters).
```

Comment:
ui element, semantic line breaks. Also can you move the information about the **Library Function Name** to its own paragraph.

```suggestion
Then you select the *Javascript Library* where you created your Javascript function, and the name of your function.
(There is no need to include the parameters.)
```